internal/database/client_intergration_test.go

```diff
@@ -0,0 +1,85 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+	hnModel "github.com/emmaLP/gs-software-onboarding/pkg/hackernews/model"
+	tc "github.com/romnn/testcontainers"
+	tcMongo "github.com/romnn/testcontainers/mongo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"go.uber.org/zap"
+)
+
+func setupMongo(ctx context.Context) (mongoC testcontainers.Container, Config tcMongo.DBConfig, err error) {
+	container, dbConfig, err := tcMongo.StartMongoContainer(ctx, tcMongo.ContainerOptions{
+		ContainerOptions: tc.ContainerOptions{CollectLogs: false},
+		User:             "test",
+		Password:         "test",
+	})
+	if err != nil {
+		return nil, tcMongo.DBConfig{}, err
+	}
+
+	return container, dbConfig, nil
+}
+
+func TestSaveItem(t *testing.T) {
+	mongo, dbConfig, err := setupMongo(context.TODO())
+	require.NoError(t, err)
+	require.NotNil(t, mongo)
+	require.NotNil(t, dbConfig)
+	defer mongo.Terminate(context.TODO())
```

Comment:
should we check for an error here?

Reply:
I've recently had this conversation with @alexdunne  around handling errors on defer calls and seem the accepted practice is not to.

Our production code is littered with examples like this one, also this been a test class I'm not too worried if that fails since a user will likely need to fix the test to pass in the first places